modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/planner/BaseStatisticsUsagePlannerTest.java

```diff
@@ -0,0 +1,289 @@
+
+    /** */
+    TestTable tbl3 = new TestTable(
+        new RelDataTypeFactory.Builder(f)
+            .add("T3C1INT", f.createJavaType(Integer.class))
+            .add("T3C2STR", f.createJavaType(String.class))
+            .add("T3C3DBL", f.createJavaType(Double.class))
+            .build()) {
+
+        @Override public ColocationGroup colocationGroup(PlanningContext ctx) {
+            return ColocationGroup.forAssignments(Arrays.asList(
+                select(nodes, 0, 1),
+                select(nodes, 1, 2),
+                select(nodes, 2, 0),
+                select(nodes, 0, 1),
+                select(nodes, 1, 2)
+            ));
+        }
+
+        @Override public IgniteDistribution distribution() {
+            return IgniteDistributions.affinity(0, "TBL3", "hash");
+        }
+    };
+
+    /**
+     * Get test tbl1 statistics.
+     *
+     * @return ObjectStatisticsImpl.
+     */
+    public ObjectStatisticsImpl getStatistics() {
+        HashMap<String, ColumnStatistics> colStat = new HashMap<>();
+        colStat.put("T1C1INT", new ColumnStatistics(ValueInt.get(1), ValueInt.get(1000),
+            0, 1000, 1000, 4, null, 1, 0));
+
+        colStat.put("T1C2STR", new ColumnStatistics(ValueString.get("A1"), ValueString.get("Z9"),
+            100, 20, 1000, 2, null, 1, 0));
+
+        colStat.put("T1C3DBL", new ColumnStatistics(ValueDouble.get(0.01), ValueDouble.get(0.99),
+            0, 1000, 1000, 8, null, 1, 0));
+
+        colStat.put("T1C4BYTE", new ColumnStatistics(ValueByte.get((byte)0), ValueByte.get((byte)255),
+            0, 1000, 1000, 8, null, 1, 0));
+
+        colStat.put("T1C5BOOLEAN", new ColumnStatistics(ValueBoolean.get(false), ValueBoolean.get(true),
+            0, 2, 1000, 1, null, 1, 0));
+
+        colStat.put("T1C6CHARACTER", new ColumnStatistics(ValueString.get("A"), ValueString.get("Z"),
+            10, 10, 1000, 1, null, 1, 0));
+
+        colStat.put("T1C7SHORT", new ColumnStatistics(ValueShort.get((short)0), ValueShort.get((short)5000),
+            110, 500, 1000, 2, null, 1, 0));
+
+        colStat.put("T1C8LONG", new ColumnStatistics(ValueLong.get(0L), ValueLong.get(100000L),
+            0, 100000, 100000, 8, null, 1, 0));
+
+        colStat.put("T1C9FLOAT", new ColumnStatistics(ValueFloat.get((float)0.1), ValueFloat.get((float)0.9),
+            0, 1000, 1000, 8, null, 1, 0));
+
+        colStat.put("T1C10DATE", new ColumnStatistics(ValueDate.get(MIN_DATE), ValueDate.get(MAX_DATE),
+            0, 1000, 1000, 8, null, 1, 0));
+
+        colStat.put("T1C11TIME", new ColumnStatistics(ValueTime.get(MIN_TIME), ValueTime.get(MAX_TIME),
+            0, 1000, 1000, 8, null, 1, 0));
+
+        colStat.put("T1C12TIMESTAMP", new ColumnStatistics(ValueTimestamp.get(MIN_TIMESTAMP), ValueTimestamp.get(MAX_TIMESTAMP),
+            0, 1000, 1000, 8, null, 1, 0));
+
+        return new ObjectStatisticsImpl(1000, colStat);
+    }
+
+    @Test
+    public void testSelectCond() throws Exception {
```

Comment:
what does it tests ?

Reply:
see ServerStatisticsIntegrationTest for tests